docs/classes/request.md

```diff
@@ -2,86 +2,128 @@
 
-## `getParam(string): string | null`
+## `addHeader(string $name, string | string[] $value)`
 
-Gets the first value of all the query string parameters with that name.
+Adds the value to the named header, or creates the header with the given value if it did not exist.
 
-## `getParamArray(string): array<string>`
+If header `$name` is not valid or `$value` is invalid, [`\Error`](http://php.net/manual/en/class.error.php) is thrown. 
 
-Gets an array with the values of the query string parameters with that name.
+{:.note}
+> To verify header `$name` and `$value` [`Message::addHeader`](message.md#addheaderstring-name-string--string-value) is used
 
-## `getAllParams(): array<array<string>>`
+## `removeHeader(string $name)`
 
-Gets the decoded query string as associative array.
+Removes the given header if it exists.
 
-## `getBody(int): Amp\ByteStream\Message`
+## `getBody(): RequestBody`
 
-Returns a representation of the request body. The [`Amp\ByteStream\Message`](http://amphp.org/byte-stream/message) can be `yield`ed to get the actual string.
+Returns the request body. The [`RequestBody`](requestbody.md) allows streamed and buffered access to an [`InputStream`](https://amphp.org/byte-stream/).
 
-The parameter `$bodySize` defaults to `-1` to take the globally configured maximum, otherwise it's the maximum body size.
+## `setBody(RequestBody | InputStream | string | null $stringOrStream)`
 
-There also exists a [`parseBody()`](parsedbody.md) function for processing of a typical HTTP form data.
+Sets the stream for the message body
 
-## `getCookie(string): string | null`
+{:.note}
+> Using a string will automatically set the `Content-Length` header to the length of the given string. Using an [`InputStream`](https://amphp.org/byte-stream/) or [`Body`](requestbody.md) instance will remove the `Content-Length` header.
 
-Gets a cookie value by name.
+If `$stringOrStream` value is not valid, [`\TypeError`](http://php.net/manual/en/class.typeerror.php) is thrown. 
 
-## `getConnectionInfo(): array`
+## `getCookies(): RequestCookie[]`
 
-Returns various information about the request, a map of the array is:
+Returns all [cookies](https://amphp.org/http/cookies) in associative map
 
-```php
-[
-    "client_port"  => int,
-    "client_addr"  => string,
-    "server_port"  => int,
-    "server_addr"  => string,
-    "is_encrypted" => bool,
-    "crypto_info"  => array, # Like returned via stream_get_meta_data($socket)["crypto"]
-]
-```
+## `getCookie(string): RequestCookie | null`
 
-## `getLocalVar(string)` / `setLocalVar(string, $value)`
+Gets a [cookie](https://amphp.org/http/cookies) value by name or `null`.
 
-These methods are only important when using [`Middleware`s](middleware.md). They manipulate the [`InternalRequest->locals`](internalrequest.html#locals) array.
+## `setCookie(RequestCookie $cookie)`
 
-## `getOption(string)`
+Adds a [`$cookie`](https://amphp.org/http/cookies) to the response.
```

Comment:
Should be `Cookie` instead of `$cookie` IMO. Also, needs to be `request` not `response`.

Reply:
Fixed. Also fixed mistake in the code doc block